inc/ruledictionnarysoftwarecollection.class.php

```diff
@@ -334,12 +365,13 @@ function replayDictionnaryOnOneSoftware(array &$new_softs, array $res_rule, $ID,
       }
 
       //Get all the different versions for a software
-      $result = $DB->query("SELECT *
-                            FROM `glpi_softwareversions`
-                            WHERE `softwares_id` = '$ID'");
+      $iterator = $DB->request([
+         'FROM'   => 'glpi_softwareversions',
+         'WHERE'  => ['softwares_id' => $ID]
+      ]);
 
-      while ($version = $DB->fetchAssoc($result)) {
-         $input["version"] = addslashes($version["name"]);
+      while ($version = $iterator->next()) {
+         $input["version"] = $version["name"];
```

Comment:
Why remove escaping ?
```suggestion
         $input["version"] = addslashes($version["name"]);
```